parens.go

```diff
@@ -8,32 +8,30 @@ import (
 
 // New returns a new root context initialised based on given options.
 func New(opts ...Option) *Context {
-	ctx := &Context{}
-	for _, opt := range withDefaults(opts) {
-		opt(ctx)
-	}
-
+	ctx := &Context{ctx: context.Background()}
 	ctx.push(stackFrame{
 		Name:          "<global>",
-		ConcurrentMap: ctx.mapFactory(),
+		ConcurrentMap: newMutexMap(),
```

Comment:
Yea, this was bit of a chicken-and-egg problem 😅 .. I was trying to add a `WithGlobals` option that would set some global vars during init. But if the frame didn't exist/initialised, this would panic. so i had to push a frame with concurrent map initialised. But definitely agreed that all frames should be using the map returned by the set map-factory.

Reply:
How about we do this:

```go
func New(opts ...Option) *Context {
        ctx := &Context{ctx: context.Background()}

	ctx.push(stackFrame{
	        Name:          "<global>",
                ConcurrentMap: newMutexMap(),

    	for _, opt := range withDefaults(opts) {
		opt(ctx)
	}

        m := ctx.mapFactory()
        for k, v := range ctx.stack[0].Native() {
                m.Store(k, v)
        }
        ctx.stack[0].ConcurrentMap = m

	return ctx
}
```

It's a bit ugly, but it's probably Good Enough (TM).

Reply:
Yea, I thought of this too (only i was planning to do it in `WithMapFactory` option). wasn't sure if there was a better way.